scanner/src/main/kotlin/scanners/ScanCode.kt

```diff
@@ -388,71 +390,101 @@ class ScanCode(name: String, config: ScannerConfiguration) : LocalScanner(name,
             try {
-                file["licenses"]?.singleOrNull() != null
-                        && LICENSE_FILE_MATCHERS.any { it.matches(Paths.get(path)) }
+                LICENSE_FILE_MATCHERS.any { it.matches(Paths.get(finding.location.path)) }
             } catch (e: InvalidPathException) {
                 false
             }
-        } ?: return ""
-
-        return getLicenseId(rootLicenseFile["licenses"].single())
-    }
+        }?.license ?: ""
 
     /**
      * Return the copyright statements in the vicinity, as specified by [toleranceLines], of [licenseStartLine] in the
      * file [path]. The default value of [toleranceLines] is set to 5 which seems to be a good balance between
      * associating findings separated by blank lines but not skipping complete license statements.
      */
     internal fun getClosestCopyrightStatements(
-        path: String,
-        copyrights: Collection<JsonNode>,
+        copyrights: List<CopyrightFinding>,
         licenseStartLine: Int,
         toleranceLines: Int = 5
     ): SortedSet<CopyrightFindings> {
+        require(copyrights.map { it.location.path }.distinct().size <= 1)
+
         val closestCopyrights = copyrights.filter {
-            (it["start_line"].intValue() - licenseStartLine).absoluteValue <= toleranceLines
+            (it.location.startLine - licenseStartLine).absoluteValue <= toleranceLines
         }
 
-        // While ScanCode 2.9.2 was still using "statements", version 2.9.7 is using "value".
-        return closestCopyrights.flatMap {
-            val startLine = it["start_line"].intValue()
-            val endLine = it["end_line"].intValue()
-            (it["statements"] ?: listOf(it["value"])).map { statements ->
-                CopyrightFindings(statements.textValue(), sortedSetOf(TextLocation(path, startLine, endLine)))
-            }
-        }.toSortedSet()
+        return closestCopyrights.map { it.toCopyrightFindings() }.toSortedSet()
     }
 
+    private fun CopyrightFinding.toCopyrightFindings() =
+        CopyrightFindings(
+            statement = statement,
+            locations = sortedSetOf(location)
+        )
+
     /**
      * Associate copyright findings to license findings within a single file.
      */
     private fun associateFileFindings(
-        path: String,
-        licenses: Collection<JsonNode>,
-        copyrights: Collection<JsonNode>,
+        licenses: List<LicenseFinding>,
+        copyrights: List<CopyrightFinding>,
         rootLicense: String = ""
     ): SortedMap<String, MutableSet<CopyrightFindings>> {
-        val copyrightsForLicenses = sortedMapOf<String, MutableSet<CopyrightFindings>>()
-
-        val allCopyrightStatements = copyrights.flatMap {
-            val startLine = it["start_line"].intValue()
-            val endLine = it["end_line"].intValue()
+        require(licenses.map { it.location.path }.distinct().size <= 1)
+        require(copyrights.map { it.location.path }.distinct().size <= 1)
```

Comment:
These will throw `IllegalArgumentException`s if the condition is false. Is this intended? If so, please add meaningful messages instead of the default "Failed requirement".

Reply:
see above.